src/signalrclient/json_hub_protocol.cpp

```diff
@@ -22,7 +22,7 @@ namespace signalr
         auto pos = message.find(record_separator, offset);
         while (pos != std::string::npos)
         {
-            auto hub_message = parse_message(message.substr(offset, pos - offset));
+            auto hub_message = parse_message(message.c_str() + offset, pos - offset);
```

Comment:
This makes me irrationally scared :). But fine.

Reply:
It did for me too, but this is just moving the c_str() call from parse_message to parse_message*s*

Reply:
But POINTERS!